lib/alertmanager_api/silence.py

```diff
@@ -0,0 +1,198 @@
+                    f"Response text: {req_ex.response.text}"
+                )
+            self.log.critical(msg)
+            raise req_ex
+        except Exception as req_ex:
+            # to capture any other Exception that is not HTTPError
+            msg = f"Failed to create silence in Alertmanager: {req_ex} "
+            self.log.critical(msg)
+            raise req_ex
+
+    def remove_silence(self, silence):
+        """
+        Removes a previously scheduled silence in alertmanager
+
+        :param silence: object including the silence_id to be removed
+        :type silence: SilenceDetails
+        :raises MissingMandatoryParamError: when the silence_id is missing
+        :raises requests.exceptions.RequestException:
+            when the request to the AlertManager failed
+        """
+        if not silence.silence_id:
+            msg = "Missing silence silence_id"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+
+        try:
+            api_url = f"{self.api_endpoint}/api/v2/silence/{silence.silence_id}"
+            response = requests.delete(
+                api_url,
+                auth=self.auth,
+                timeout=10,
+            )
+            response.raise_for_status()
+        except requests.HTTPError as req_ex:
+            msg = f"Failed to delete silence in Alertmanager: {req_ex} "
+            if req_ex.response is not None:
+                msg += (
+                    f"Failed to delete silence with ID {silence.silence_id}. "
+                    f"Response status code: {response.status_code} "
+                    f"Response text: {response.text}"
+                )
+            self.log.critical(msg)
+            raise req_ex
+        except Exception as req_ex:
+            # to capture any other Exception that is not HTTPError
+            msg = f"Failed to create silence in Alertmanager: {req_ex} "
+            self.log.critical(msg)
+            raise req_ex
+
+    def remove_silences(self, silence_instances):
+        """
+        Removes a list of previously scheduled silences in alertmanager
+
+        :param silence_instances: a list of silence instances
+        :type silence: SilenceDetailsHandler
+        """
+        for silence in silence_instances:
+            self.remove_silence(silence)
+
+    def get_silences(self):
+        """
+        download all silence events recorded in AlertManager
+
+        :return: the list of Silence events currently recorded in AlertManager
+        :rtype: SilenceDetailsHandler
+        """
+        try:
+            api_url = f"{self.api_endpoint}/api/v2/silences"
+            response = requests.get(api_url, auth=self.auth, timeout=10)
+            response.raise_for_status()
+            silences = response.json()
+            handler = SilenceDetailsHandler()
+            handler.add_from_json(silences)
+            return handler
+        except requests.HTTPError as req_ex:
```

Comment:
what's the purpose of having 2 different except statements for requests.HTTPError and RequestException if they both log the same message? We can combine into one